Mono.Cecil/BaseAssemblyResolver.cs

```diff
@@ -158,8 +158,14 @@ AssemblyDefinition SearchDirectory (AssemblyNameReference name, IEnumerable<stri
 			foreach (var directory in directories) {
 				foreach (var extension in extensions) {
 					string file = Path.Combine (directory, name.Name + extension);
-					if (File.Exists (file))
-						return GetAssembly (file, parameters);
+					if (File.Exists (file)) {
+						try {
+							return GetAssembly (file, parameters);
+						}
+						catch (System.BadImageFormatException) {
```

Comment:
In the Mono coding conventions, the catch goes on the same line as the closing } for the try, like this:

```csharp
} catch {
```